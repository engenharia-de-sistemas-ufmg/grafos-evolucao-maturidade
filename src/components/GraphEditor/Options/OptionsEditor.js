import React from 'react';
import PropTypes from 'prop-types';
import { optionsType } from '../../../utils/types';
import OptionsAce from './OptionsAce';

const OptionsEditor = ({ lang, options, updateOptions }) => {
  const update = (data) => {
    updateOptions(data);
  };
  return (
    <div className="columns">
      <div className="column is-half">
        <OptionsAce lang={lang} options={options} update={update} />
      </div>
    </div>
  );
};

OptionsEditor.propTypes = {
  lang: PropTypes.string.isRequired,
  options: optionsType.isRequired,
  updateOptions: PropTypes.func.isRequired,
};

export default OptionsEditor;
