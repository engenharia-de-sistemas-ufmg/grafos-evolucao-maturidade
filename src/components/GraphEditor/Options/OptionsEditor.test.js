import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import OptionsEditor from './OptionsEditor';

describe('OptionsEditor', () => {
  Enzyme.configure({ adapter: new Adapter() });

  test('should match snapshot', () => {
    const component = shallow(
      <OptionsEditor
        lang="pt"
        options={{ hierarchical: true, animation: false }}
        updateOptions={() => { }}
      />,
    );
    expect(toJson(component)).toMatchSnapshot();
  });
});
