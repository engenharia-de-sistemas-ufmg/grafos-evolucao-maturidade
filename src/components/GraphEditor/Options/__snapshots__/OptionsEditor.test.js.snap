// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OptionsEditor should match snapshot 1`] = `
<div
  className="columns"
>
  <div
    className="column is-half"
  >
    <OptionsAce
      lang="pt"
      options={
        Object {
          "height": "500px",
        }
      }
      update={[Function]}
    />
  </div>
</div>
`;
